### PR TITLE
TEAM TEACHING - merge from master

### PR DESCRIPTION
**Pull request by ZappoMan**

### CODE REVIEW

File: libraries/entities-renderer/src/RenderableLineEntityItem.cpp [hunk 1]
```diff
@@ -29,10 +30,26 @@ void RenderableLineEntityItem::render(RenderArgs* args) {
     glm::vec3 p1 = ENTITY_ITEM_ZERO_VEC3;
     glm::vec3 p2 = getDimensions();
     glm::vec4 lineColor(toGlm(getXColor()), getLocalRenderAlpha());
+
     Q_ASSERT(args->_batch);
     gpu::Batch& batch = *args->_batch;
     batch.setModelTransform(getTransformToCenter());
-    DependencyManager::get<DeferredLightingEffect>()->renderLine(batch, p1, p2, lineColor, lineColor);
+
```
Comment (ZappoMan) [inline libraries/entities-renderer/src/RenderableLineEntityItem.cpp]: note: @Atlante45 this part of the merge had a conflict... can you review to see if I got it right? you're transform changes vs eric's new poly line code.

File: libraries/entities-renderer/src/RenderablePolyVoxEntityItem.cpp [hunk 2]
```diff
@@ -33,22 +33,63 @@
...
+
+    setVoxelVolumeSize(_voxelVolumeSize);
+}
+
 RenderablePolyVoxEntityItem::~RenderablePolyVoxEntityItem() {
     delete _volData;
 }
 
-void RenderablePolyVoxEntityItem::setVoxelVolumeSize(glm::vec3 voxelVolumeSize) {
+bool inUserBounds(const PolyVox::SimpleVolume<uint8_t>* vol, PolyVoxEntityItem::PolyVoxSurfaceStyle surfaceStyle,
```
Comment (ZappoMan) [inline libraries/entities-renderer/src/RenderablePolyVoxEntityItem.cpp]: note: this does appear to be the same change made in master

File: libraries/entities-renderer/src/RenderablePolyVoxEntityItem.cpp [hunk 3]
```diff
@@ -237,33 +441,44 @@ bool RenderablePolyVoxEntityItem::findDetailedRayIntersection(const glm::vec3& o
         return true;
     }
 
+    // the PolyVox ray intersection code requires a near and far point.
```
Comment (ZappoMan) [inline libraries/entities-renderer/src/RenderablePolyVoxEntityItem.cpp]: this section had lots of conflicts... I tried to keep the master code but replace the _dimensions with getDimensions() etc.

File: libraries/entities-renderer/src/RenderableLineEntityItem.cpp (hunk 1 above)
Comment (Atlante45) [inline libraries/entities-renderer/src/RenderableLineEntityItem.cpp]: looks correct, I just need to update the rendering.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7c40b666e8e8cd6255676065135a501593a5ad0d available at: https://gist.github.com/606762d42045efb5c87a<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5032.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5032.dmg

**Pull request merged**